crates/sound/src/audio_backends/null.rs

```diff
@@ -1,30 +1,29 @@
 // SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
 
+use std::sync::{Arc, RwLock};
+
 use super::AudioBackend;
-use crate::Result;
+use crate::{Result, Stream};
 
-pub struct NullBackend {}
+pub struct NullBackend {
+    streams: Arc<RwLock<Vec<Stream>>>,
+}
 
 impl NullBackend {
-    pub fn new() -> Self {
-        NullBackend {}
+    pub fn new(streams: Arc<RwLock<Vec<Stream>>>) -> Self {
+        Self { streams }
     }
 }
 
 impl AudioBackend for NullBackend {
     fn write(&self, stream_id: u32) -> Result<()> {
-        println!("null backend, writting to stream: {}", stream_id);
+        log::trace!("NullBackend write stream_id {}", stream_id);
+        _ = std::mem::take(&mut self.streams.write().unwrap()[stream_id as usize].buffers);
         Ok(())
     }
 
-    fn read(&self, _stream_id: u32) -> Result<()> {
-        /*
-        let buf = req.data_slice().ok_or(Error::SoundReqMissingData)?;
-        let zero_mem = vec![0u8; buf.len()];
-
-        buf.copy_from(&zero_mem);
-
-        */
+    fn read(&self, _id: u32) -> Result<()> {
```

Comment:
```suggestion
    fn read(&self, id: u32) -> Result<()> {
```
as now this variable is "used"

Reply:
Used just for `log::trace!`, doesn't make any difference.